x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/InferenceStats.java

```diff
@@ -194,73 +208,59 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeVLong(this.missingAllFieldsCount);
         out.writeVLong(this.inferenceCount);
         out.writeVLong(this.failureCount);
+        if (out.getVersion().onOrAfter(Version.V_7_9_0)) {
+            out.writeVLong(this.cacheMissCount);
+        }
         out.writeOptionalString(this.modelId);
         out.writeOptionalString(this.nodeId);
         out.writeInstant(timeStamp);
     }
 
     public static class Accumulator {
 
-        private final LongAdder missingFieldsAccumulator = new LongAdder();
-        private final LongAdder inferenceAccumulator = new LongAdder();
-        private final LongAdder failureCountAccumulator = new LongAdder();
+        private long missingFieldsAccumulator = 0L;
+        private long inferenceAccumulator = 0L;
+        private long failureCountAccumulator = 0L;
+        private long cacheMissAccumulator = 0L;
         private final String modelId;
         private final String nodeId;
-        // curious reader
-        // you may be wondering why the lock set to the fair.
-        // When `currentStatsAndReset` is called, we want it guaranteed that it will eventually execute.
-        // If a ReadWriteLock is unfair, there are no such guarantees.
-        // A call for the `writelock::lock` could pause indefinitely.
-        private final ReadWriteLock readWriteLock = new ReentrantReadWriteLock(true);
-
-        public Accumulator(String modelId, String nodeId) {
+
+        public Accumulator(String modelId, String nodeId, long cacheMisses) {
             this.modelId = modelId;
             this.nodeId = nodeId;
+            this.cacheMissAccumulator = cacheMisses;
         }
 
-        public Accumulator(InferenceStats previousStats) {
+        Accumulator(InferenceStats previousStats) {
             this.modelId = previousStats.modelId;
             this.nodeId = previousStats.nodeId;
-            this.missingFieldsAccumulator.add(previousStats.missingAllFieldsCount);
-            this.inferenceAccumulator.add(previousStats.inferenceCount);
-            this.failureCountAccumulator.add(previousStats.failureCount);
+            this.missingFieldsAccumulator += previousStats.missingAllFieldsCount;
+            this.inferenceAccumulator += previousStats.inferenceCount;
+            this.failureCountAccumulator += previousStats.failureCount;
+            this.cacheMissAccumulator += previousStats.cacheMissCount;
         }
 
         public Accumulator merge(InferenceStats otherStats) {
```

Comment:
I think this method needs to be `synchronized` too

Reply:
It is only used within `ConcurrentHashMap#compute`
Some docs
```
The entire method invocation is performed atomically. 
Some attempted update operations on this map by other threads
may be blocked while computation is in progress, so the
computation should be short and simple, and must not attempt to
update any other mappings of this Map.
```
I will add a comment saying this method is not thread safe itself and should be used in a thread safe fashion.